src/activity_handlers.cpp

```diff
@@ -696,6 +708,18 @@ void activity_handlers::fill_liquid_do_turn( player_activity *act_, player *p )
             on_ground = source_stack.begin();
             std::advance( on_ground, act.values.at( 1 ) );
             liquid = *on_ground;
+			break;
+        case LST_MONSTER:
+            Creature *c = g->critter_at( source_pos );
+            source_mon = dynamic_cast<monster *>( c );
+            if( source_mon == nullptr ) {
+                debugmsg( "could not find source creature for liquid transfer" );
```

Comment:
I don't fully(or even partially) understand how player_activity works, I thought that `act.set_to_null();` would simple stop the activity.

There was a `void finish( player *p );` but doesn't seems to have a definition and is private.

If you mind to please explain how player_activity and activity_id work I ill try to find a better way, otherwise I don't mind putting the throw back. I really don't have any idea what would be the most proper and correct way to handle this situation so any help is welcomed.